### PR TITLE
feat(k3s): add memory knobs to harness_k3s

### PR DESCRIPTION
**Pull request by stormqueen1990**
* Include an option for the user to specify memory resources for the k3s harness resource.

### CODE REVIEW

File: internal/provider/harness.go [hunk 1]
```diff
@@ -4,13 +4,15 @@ import (
...
 
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/containers/provider"
 	"github.com/chainguard-dev/terraform-provider-imagetest/internal/inventory"
 	"github.com/chainguard-dev/terraform-provider-imagetest/internal/log"
 	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
 	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	resource2 "k8s.io/apimachinery/pkg/api/resource"
```
Comment (joshrwolf) [inline internal/provider/harness.go:15]:
```suggestion
	kresource "k8s.io/apimachinery/pkg/api/resource"
```

nit!

File: internal/provider/harness.go [hunk 2]
```diff
@@ -158,6 +160,40 @@ func (r *HarnessResource) ShouldSkip(ctx context.Context, req resource.CreateReq
 	return skip
 }
 
+// ParseMemoryResources parses memory resources for harnesses, returning a ContainerResourcesRequest object if parsing
+// is successful or an error if values are not parseable.
+func ParseMemoryResources(memoryResources ContainerMemoryResources) (*provider.ContainerResourcesRequest, error) {
```
Comment (joshrwolf) [inline internal/provider/harness.go:165]: can we plumb through a `ParseCpuResources` in a similar way? that was the primary knob I was hoping to use for the pytorch tests 🙏 

if it warrants a follow up PR as to not make this overly messy that's fine too!

Comment (stormqueen1990) [reply internal/provider/harness.go:165]: I was planning to originally, but the Docker API is a little strange in that regard: https://docs.docker.com/config/containers/resource_constraints/#cpu

I'm really unsure which one of those would be the request vs limit, do you have any insights on this?

File: internal/provider/harness.go (hunk 1 above)
Comment (stormqueen1990) [reply internal/provider/harness.go]: Updated!

**Review by joshrwolf — APPROVED** [review]

File: internal/provider/harness.go (hunk 2 above)
Comment (joshrwolf) [reply internal/provider/harness.go:165]: my guess is its `--cpus` for the request, and `--cpu-quota` for the limits. either way, I approved this, will let you decide if you want to include it in a follow up or not

Comment (stormqueen1990) [reply internal/provider/harness.go:165]: Opened #98 to follow-up

**Pull request merged**